[
    {
        "type": "message",
        "user": "U1AU8DRQR",
        "text": "<https:\/\/gerrit.hyperledger.org\/r\/#\/c\/2905\/1>",
        "ts": "1480584968.004548"
    },
    {
        "type": "message",
        "user": "U1SHFS1Q9",
        "text": "Can someone provide clarification on how the Validating Leader is selected or elected during consensus ? Thank you",
        "ts": "1480586375.004549"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "Hi. I am using fabric v 0.6. I have set up a network of 4 peers and 1 CA. I am trying to connect the Non-Validating peer to the network, When i try to use the \/chain Rest API on the NVP it gives me {\"Error\":\"No blocks in blockchain.\"}",
        "ts": "1480588052.004550"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "<@U12AHQ3NK> are all vp's and nvp's are on same network ?",
        "ts": "1480588195.004551"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "<@U1F1CU61Y>  yes",
        "ts": "1480588819.004552"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "i am getting the same problem",
        "ts": "1480588840.004553"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "what i am doubting is that every peer is listening on 172.17.0.2",
        "ts": "1480588861.004554"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "they are not able to connect through physical ip of machine",
        "ts": "1480588880.004555"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "so nvp must forward invoke\/query request to vp and it tries to connect 172.17.0.2 that is it gets connected to self",
        "ts": "1480588922.004556"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "what is the command you using to run a NVP?",
        "ts": "1480588942.004557"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "peer node start",
        "ts": "1480588958.004558"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "just change in peer\/core.yaml, set value of validator=false",
        "ts": "1480588980.004559"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "what about you",
        "ts": "1480588987.004560"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "same",
        "ts": "1480589036.004561"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "so i am expecting this problem",
        "ts": "1480589052.004562"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "what do you say ?",
        "ts": "1480589062.004563"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "<@U12AHQ3NK> -  a non-validating peer is really no more than a glorified \"wallet\" for submitting transactions to validating peers.    moving forward, we don't even have non-validating peers in the v1 architecture.  What is your goal in terms of using an NVP?",
        "ts": "1480590976.004564"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "<@U0PB67X4K>  in my case i want to connect a third party system to nvp so that third party may not be able to deploy chaincode as well he has replica of that data in his local network and he can read it fastly",
        "ts": "1480591376.004565"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "<@U0PB67X4K>: my goal is to connect some peers to the network and get the blockchain data. ",
        "ts": "1480591390.004566"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "<@U1F1CU61Y> - not going to work - the only way to query data is via chaincode and chaincode is not deployed to non-validating peers",
        "ts": "1480591636.004567"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "<@U12AHQ3NK> - why not just get the blockchain data from an application using the SDK?   Why does it need to be a NVP?",
        "ts": "1480591698.004568"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "<@U0PB67X4K> one question from my side. Is this possible, we give  permissions to particular users so that only they can deploy chaincode",
        "ts": "1480591748.004569"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "means is there any access right management available ?",
        "ts": "1480591780.004570"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "not in v0.6",
        "ts": "1480591802.004571"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "<@U0PB67X4K>:  can I add new peers to the network dynamically in v 0.6?",
        "ts": "1480591815.004572"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "then it may be security issues that if i give a peer to third party, he may deploy chaincode",
        "ts": "1480591857.004573"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "and read the data written by another chaincode too ?? am i right or not ?",
        "ts": "1480591890.004574"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "<@U12AHQ3NK> - no - you cannot add validating peers dynamically in v0.6",
        "edited": {
            "user": "U0PB67X4K",
            "ts": "1480592145.000000"
        },
        "ts": "1480591891.004575"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "<@U1F1CU61Y> - you can build access control into your chaincode methods themselves to prevent certain clients from invoking or querying a specific chaincode or specific functions on that chaincode",
        "ts": "1480591962.004576"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "<@U0PB67X4K>: through SDK ?",
        "ts": "1480591971.004577"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "<@U0PB67X4K>  yes that we can do. suppose if i have written some data with chaincode1. can we read that data with chaincode2?",
        "ts": "1480592020.004578"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "you can't really stop someone from deploying chaincode in fabric v0.6 without putting some type of \"proxy\" layer in between",
        "ts": "1480592023.004579"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "with same key",
        "ts": "1480592025.004580"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "<@U1F1CU61Y> - the data itself is scoped at a chaincode level so if you restrict access to chaincode within your functions then you would not be able to invoke those functions from other chaincode unless you also had permission to access the initial chaincode",
        "ts": "1480592128.004581"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "<@U12AHQ3NK> -  sorry - had a type - you CANNOT dynamically add validating peers to the network in v0.6",
        "ts": "1480592171.004583"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "<@U0PB67X4K>: I have a network of 5 VP's. I connected 4 peers first and then deployed a chaincode. Later I connected one more peer to the network. I noticed tht the new peer didn't sync the blocks. i am using v0.6",
        "ts": "1480592371.004584"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "that's the expected behavior.  there's a slight change that if you stop your original 4 peers, modify the config to say that there are 5 peers and restart that things *might* work, but v0.6 does not support dynamically adding a peer to a network",
        "ts": "1480592574.004585"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "<@U0PB67X4K>:  I started initially wiht N=5 then started 4 peers initially",
        "ts": "1480592629.004586"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "<@U0PB67X4K>  so the only way to add new peers is to stop the peers, change config and then restart them?",
        "ts": "1480592740.004587"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "hmmm - if you specified N=5 to start with, you should be able to start up the 5th peer after the fact.  But you'd have to keep running transactions for the 5th peer to notice it is behind and catch up.   As I recall, you'd probably need to generate between 10 and 20 blocks of transactions",
        "ts": "1480592986.004588"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "<@U0PB67X4K>: so I should be generating 10-20 transactions into the network ?",
        "ts": "1480593178.004589"
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "something like that - I forgot the default block size and timeout, but I think its something low like 2 transactions \/ block?",
        "ts": "1480593652.004590"
    },
    {
        "type": "message",
        "user": "U12AHQ3NK",
        "text": "<@U0PB67X4K>:  thanks..i wil try that out ",
        "ts": "1480593852.004591"
    },
    {
        "type": "message",
        "user": "U1F1CU61Y",
        "text": "<@U0PB67X4K> thanks for your support",
        "ts": "1480598575.004592"
    },
    {
        "type": "message",
        "user": "U1SHFS1Q9",
        "text": "<@U0PB67X4K>, You mentioned in your earlier post that only way to query data is via chaincode and chaincode is not deployed to non-validating peers.\n\nHowever, the spec (<https:\/\/github.com\/hyperledger\/fabric\/blob\/master\/docs\/protocol-spec.md#222-multiple-validating-peers>) says,\n\n\"Non validating peers (also known as peers) receive user transactions on behalf of users, and after some fundamental validity checks, they forward the transactions to their neighboring validating peers. Peers maintain an up-to-date copy of the blockchain, but in contradiction to validators, they do not execute transactions (a process also known as transaction validation).\u201d\n\nIf it has an upto data copy of the blockchain, we should be able to query data, right? or my understanding is wrong?",
        "attachments": [
            {
                "service_name": "GitHub",
                "title": "hyperledger\/fabric",
                "title_link": "https:\/\/github.com\/hyperledger\/fabric\/blob\/master\/docs\/protocol-spec.md#222-multiple-validating-peers",
                "text": "Read-only mirror of <https:\/\/gerrit.hyperledger.org\/r\/#\/admin\/projects\/fabric>",
                "fallback": "GitHub: hyperledger\/fabric",
                "thumb_url": "https:\/\/avatars3.githubusercontent.com\/u\/7657900?v=3&s=400",
                "from_url": "https:\/\/github.com\/hyperledger\/fabric\/blob\/master\/docs\/protocol-spec.md#222-multiple-validating-peers",
                "thumb_width": 142,
                "thumb_height": 142,
                "service_icon": "https:\/\/github.com\/apple-touch-icon.png",
                "id": 1
            }
        ],
        "ts": "1480599895.004593"
    },
    {
        "type": "message",
        "user": "U0UGH3X7X",
        "text": "scrum ...",
        "ts": "1480604416.004595"
    },
    {
        "text": "<@U0UGH3X7X|tuand> has started a Google+ Hangout for this channel. <https:\/\/hangouts.google.com\/hangouts\/_\/r6xhuz4wsvfxnarynsnr25zaxqe|Join Hangout>.",
        "username": "hangouts",
        "bot_id": "B0UKUAJ0Y",
        "type": "message",
        "subtype": "bot_message",
        "ts": "1480604438.004596"
    },
    {
        "type": "message",
        "user": "U0UGH3X7X",
        "text": "<@U1KDAMDJ7>  trying to run kafka docker-compose ( from 2459 change set ) on OSX\/vagrant ... failing with `ERROR: In file '.\/docker-compose.yml' service 'version' doesn't have any configuration options. All top level keys in your docker-compose.yml must map to a dictionary of configuration options.`",
        "ts": "1480611541.004597"
    },
    {
        "type": "message",
        "user": "U1KDAMDJ7",
        "text": "you might have an old docker-compose installed",
        "ts": "1480611576.004598"
    },
    {
        "type": "message",
        "user": "U0UGH3X7X",
        "text": "i think it's because my docker-compose version is backlevel ?  1.5.2 , need to be at 1.6 ?",
        "ts": "1480611578.004599"
    },
    {
        "type": "message",
        "user": "U1KDAMDJ7",
        "text": "Yes",
        "ts": "1480611596.004600"
    },
    {
        "type": "message",
        "user": "U0UGH3X7X",
        "text": "you running inside vagrant ?",
        "ts": "1480611603.004601"
    },
    {
        "type": "message",
        "user": "U1KDAMDJ7",
        "text": "Yes, but it should work outside also.",
        "ts": "1480611643.004602"
    },
    {
        "type": "message",
        "user": "U1KDAMDJ7",
        "text": "<https:\/\/docs.docker.com\/docker-for-mac\/>",
        "attachments": [
            {
                "service_name": "Docker",
                "title": "Get started with Docker for Mac",
                "title_link": "https:\/\/docs.docker.com\/docker-for-mac\/",
                "text": "Getting Started",
                "fallback": "Docker: Get started with Docker for Mac",
                "image_url": "https:\/\/www.docker.com\/sites\/default\/files\/social\/docker-twitter-share.png",
                "ts": 1480558148,
                "from_url": "https:\/\/docs.docker.com\/docker-for-mac\/",
                "image_width": 625,
                "image_height": 250,
                "image_bytes": 49214,
                "service_icon": "https:\/\/docs.docker.com\/favicons\/apple-touch-icon-57x57.png",
                "id": 1
            }
        ],
        "ts": "1480611643.004603"
    },
    {
        "type": "message",
        "subtype": "file_share",
        "text": "<@U1KDAMDJ7|sanchezl> uploaded a file: <https:\/\/hyperledgerproject.slack.com\/files\/sanchezl\/F38MZTF9N\/pasted_image_at_2016_12_01_12_01_pm.png|Pasted image at 2016-12-01, 12:01 PM>",
        "file": {
            "id": "F38MZTF9N",
            "created": 1480611677,
            "timestamp": 1480611677,
            "name": "Pasted image at 2016_12_01 12_01 PM.png",
            "title": "Pasted image at 2016-12-01, 12:01 PM",
            "mimetype": "image\/png",
            "filetype": "png",
            "pretty_type": "PNG",
            "user": "U1KDAMDJ7",
            "editable": false,
            "size": 1983,
            "mode": "hosted",
            "is_external": false,
            "external_type": "",
            "is_public": true,
            "public_url_shared": false,
            "display_as_bot": false,
            "username": "",
            "url_private": "https:\/\/files.slack.com\/files-pri\/T0J024XGA-F38MZTF9N\/pasted_image_at_2016_12_01_12_01_pm.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "url_private_download": "https:\/\/files.slack.com\/files-pri\/T0J024XGA-F38MZTF9N\/download\/pasted_image_at_2016_12_01_12_01_pm.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_64": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F38MZTF9N-c2c872ae48\/pasted_image_at_2016_12_01_12_01_pm_64.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_80": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F38MZTF9N-c2c872ae48\/pasted_image_at_2016_12_01_12_01_pm_80.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_360": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F38MZTF9N-c2c872ae48\/pasted_image_at_2016_12_01_12_01_pm_360.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "thumb_360_w": 43,
            "thumb_360_h": 41,
            "thumb_160": "https:\/\/files.slack.com\/files-tmb\/T0J024XGA-F38MZTF9N-c2c872ae48\/pasted_image_at_2016_12_01_12_01_pm_160.png?t=xoxe-18002167554-139099126023-137701436192-e599afc92e",
            "image_exif_rotation": 1,
            "original_w": 43,
            "original_h": 41,
            "permalink": "https:\/\/hyperledgerproject.slack.com\/files\/sanchezl\/F38MZTF9N\/pasted_image_at_2016_12_01_12_01_pm.png",
            "permalink_public": "https:\/\/slack-files.com\/T0J024XGA-F38MZTF9N-19f5c2d8a3",
            "channels": [
                "C0Z4NBUN6",
                "C27E2LGJY"
            ],
            "groups": [],
            "ims": [],
            "comments_count": 0
        },
        "user": "U1KDAMDJ7",
        "upload": true,
        "display_as_bot": false,
        "username": "<@U1KDAMDJ7|sanchezl>",
        "bot_id": null,
        "ts": "1480611678.004605"
    },
    {
        "type": "message",
        "user": "U0UGH3X7X",
        "text": "i'm running inside vagrant and getting the error  ....  time to rebuild my image :slightly_smiling_face:",
        "ts": "1480611683.004606"
    },
    {
        "type": "message",
        "user": "U1KDAMDJ7",
        "text": "Do you have that icon",
        "ts": "1480611684.004607"
    },
    {
        "type": "message",
        "user": "U1KDAMDJ7",
        "text": "ahh, yes\u2026 you\u2019ll need to pick up the updated compose",
        "ts": "1480611703.004608"
    },
    {
        "type": "message",
        "user": "U0UGH3X7X",
        "text": "thx ! onward ...",
        "ts": "1480611721.004609",
        "reactions": [
            {
                "name": "slightly_smiling_face",
                "users": [
                    "U1KDAMDJ7"
                ],
                "count": 1
            }
        ]
    },
    {
        "type": "message",
        "user": "U0PB67X4K",
        "text": "@tuan - best to move to compose 1.8",
        "ts": "1480612772.004610"
    },
    {
        "user": "U11BD9FA7",
        "text": "<@U11BD9FA7|ynamiki> has left the channel",
        "type": "message",
        "subtype": "channel_leave",
        "ts": "1480640199.004611"
    }
]